compiler/natives/src/crypto/sha256/sha256.go

```diff
@@ -0,0 +1,67 @@
+// +build js
+
+package sha256
+
+import (
+	"errors"
+
+	"github.com/gopherjs/gopherjs/js"
+)
+
+var ErrWebCryptoFailed = errors.New("crypto: Web Crypto API call failed")
+
+// Get the SubtleCrypto interface
+// https://www.w3.org/TR/WebCryptoAPI/
+func getSubtle() *js.Object {
+	crypto := js.Global.Get("crypto")
+	if crypto == js.Undefined {
+		crypto = js.Global.Get("msCrypto") // for IE11
+		if crypto == js.Undefined {
+			return nil
+		}
+	}
+	subtle := crypto.Get("subtle")
+	if subtle == js.Undefined {
+		subtle = crypto.Get("webkitSubtle") // for Safari
+		if subtle == js.Undefined {
+			return nil
+		}
+	}
+	return subtle
+}
+
+func jsSum256(data []byte) (res [Size]byte, err error) {
+	subtle := getSubtle()
+	if subtle == nil {
+		return res, ErrWebCryptoFailed
+	}
+
+	algo := js.M{
+		"name": "SHA-256",
+	}
+	resCh := make(chan []byte, 1)
+	errCh := make(chan error, 1)
+	promise := subtle.Call("digest", algo, data)
+	promise.Call("then", func(result *js.Object) {
+		resCh <- js.Global.Get("Uint8Array").New(result).Interface().([]byte)
+	})
+	promise.Call("catch", func(*js.Object) {
+		errCh <- ErrWebCryptoFailed
+	})
+	select {
+	case err = <-errCh:
+	case resSlice:= <-resCh:
+		copy(res[:], resSlice)
+	}
+	return
+}
+
+func Sum256(data []byte) [Size]byte {
+	res, err := jsSum256(data)
+	if err != nil {
+		// The WebCryptoAPI call failed: fallback by calling the implementation from
+		// the Go standard library
+		return _gopherjs_overriden_Sum256(data)
+	}
```

Comment:
5ms doesn't sound like enough to worry about to me. At least not at this stage.  Micro-optimizations like that should come much later :)